rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -141,7 +141,32 @@ public String getResourceGroupId() {
     @Override
     public String getResourceId() {
         if (jdbcUrl.contains("?")) {
-            return jdbcUrl.substring(0, jdbcUrl.indexOf('?'));
+            StringBuilder jdbcUrlBuilder = new StringBuilder();
+            jdbcUrlBuilder.append(jdbcUrl.substring(0, jdbcUrl.indexOf('?')));
+            /*
+             * prevent pg sql url like
+             * jdbc:postgresql://127.0.0.1:5432/seata?currentSchema=public
+             * jdbc:postgresql://127.0.0.1:5432/seata?currentSchema=seata
+             * cause the duplicated resourceId
+             * it will cause the problem like
+             * 1.get file lock fail
+             * 2.error table meta cache
+             */
+            StringBuilder paramsBuilder = new StringBuilder();
+            String paramUrl = jdbcUrl.substring(jdbcUrl.indexOf('?') + 1, jdbcUrl.length());
+            String[] urlParams = paramUrl.split("&");
+            for (String urlParam : urlParams) {
+                if (urlParam.contains("currentSchema")) {
+                    paramsBuilder.append(urlParam);
+                }
```

Comment:
1.currentSchema is magic number
2.break?

Reply:
ok. Already add break